File: go/leetcode/283. Move Zeroes.go
func moveZeroes(nums []int)  {
    anchor := 0
    explorer := 0
    
    for explorer < len(nums) {
        
        if nums[explorer] != 0 {
            
            if explorer != anchor {
                temp := nums[explorer]
                nums[explorer] = nums[anchor]
                nums[anchor] = temp
            }
            
            anchor++
        }
        
        explorer++
    }
}

File: javascript/leetcode/283. Move Zeroes.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function(nums) {
  let anchor = 0;
  let explorer = 0;

  while (explorer < nums.length) {
    if (nums[explorer] !== 0) {
      if (explorer !== anchor) {
        let temp = nums[explorer];
        nums[explorer] = nums[anchor];
        nums[anchor] = temp;
      }

      anchor++;
    }

    explorer++;
  }
};
